custom/enikshay/tests/test_adherence_updates.py

```diff
@@ -0,0 +1,363 @@
+                'adherence_total_doses_taken': 0
+            }
+        )
+
+    def test_adherence_date_less_than_purge_date(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 20),
+                (datetime(2016, 1, 10), 'schedule1'),
+                [(datetime(2016, 1, 15), DOSE_TAKEN_INDICATORS[0])]
+            ),
+            {
+                'aggregated_score_date_calculated': date(2016, 1, 15),
+                'expected_doses_taken': (5.0 / 7) * int(self.fixture_data['schedule1']),
+                'aggregated_score_count_taken': 1,
+                'adherence_latest_date_recorded': date(2016, 1, 15),
+                'adherence_total_doses_taken': 1
+            }
+        )
+
+    def test_adherence_date_greater_than_purge_date(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 20),
+                (datetime(2016, 1, 10), 'schedule1'),
+                [(datetime(2016, 1, 22), DOSE_TAKEN_INDICATORS[0])]
+            ),
+            {
+                'aggregated_score_date_calculated': date(2016, 1, 20),
+                'expected_doses_taken': (10.0 / 7) * int(self.fixture_data['schedule1']),
+                'aggregated_score_count_taken': 0,
+                'adherence_latest_date_recorded': date(2016, 1, 22),
+                'adherence_total_doses_taken': 1
+            }
+        )
+
+    def test_multiple_adherence_cases_all_greater(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 20),
+                (datetime(2016, 1, 10), 'schedule1'),
+                [
+                    # same day, different time
+                    (datetime(2016, 1, 21, 1), DOSE_TAKEN_INDICATORS[0]),
+                    (datetime(2016, 1, 21, 3), 'unknown'),
+                    (datetime(2016, 1, 22), DOSE_TAKEN_INDICATORS[0]),
+                    (datetime(2016, 1, 24), DOSE_TAKEN_INDICATORS[0]),
+                ]
+            ),
+            {
+                'aggregated_score_date_calculated': date(2016, 1, 20),
+                'expected_doses_taken': (10.0 / 7) * int(self.fixture_data['schedule1']),
+                'aggregated_score_count_taken': 0,
+                'adherence_latest_date_recorded': date(2016, 1, 24),
+                'adherence_total_doses_taken': 3
+            }
+        )
+
+    def test_multiple_adherence_cases_all_less(self):
+        self.assert_update(
+            (
+                datetime(2016, 1, 20),
+                (datetime(2016, 1, 10), 'schedule1'),
+                [
+                    # same day, different time. Set hours different so that case-id becomes different
+                    (datetime(2016, 1, 11, 1), DOSE_TAKEN_INDICATORS[0]),
+                    (datetime(2016, 1, 11, 3), 'unknown'),
+                    (datetime(2016, 1, 12), DOSE_TAKEN_INDICATORS[0]),
+                    (datetime(2016, 1, 14), 'unknown'),
```

Comment:
reminder that we want to use whatever we set up as the _missing_data thing instead of unknown, just for consistency with what will be coming in from the device.